src/python/WMCore/WorkQueue/WMBSHelper.py

```diff
@@ -676,11 +676,17 @@ def _addACDCFileToWMBSFile(self, acdcFile, inFileset=True):
         adds the ACDC files into WMBS database
         """
         wmbsParents = []
-        for parent in acdcFile["parents"]:
-            parent = self._addACDCFileToWMBSFile(DatastructFile(lfn=parent,
-                                                                locations=acdcFile["locations"]),
-                                                 inFileset=False)
-            wmbsParents.append(parent)
+        # TODO:  this check can be removed when ErrorHandler filters parents file for unmerged data
+        if acdcFile.get("merged", 0) and len(acdcFile["parents"]) and "/umerged/" in acdcFile["parents"][0]:
```

Comment:
In plain english: if input file is unmerged and there are unmerged parents, skip it. Is it correct? Maybe you can add this comment here just so I don't ask you again :-)

In addition to that, I'd replace the last check by `and acdcFile["parents"][0].startswith("/store/unmerged")`, just to avoid any possible "unmerged" somewhere in the file name.

Reply:
actually the condition says 
If input filed is "merged" and parents are unmerged we skip it. I will update the comments.
Ah, so it is always /store/unmerged?  I was thinking about that but not sure whether /store/ can be changed later so that is why I did /unmerged/ instead of just unmerged.
Anyway I will update that.